### PR TITLE
Test case to exercise vega-lite schema generation #671

### PR DESCRIPTION
**Pull request by sramam**
An attempt to tackle #671

@domoritz, I've added the basic scaffolding to make this happen. 
The test case is however failing. I suspect it's some dev dependency that allows it to pass when run within the context of the repo
but not otherwise. I'd appreciate your help in finding/fixing this issue. 

Doing so will hopefully nudge the coverage numbers for #666 upwards. Outside which, #666 seems good to go.

### CODE REVIEW

**Comment by domoritz** [comment]
```patch
diff --git a/package.json b/package.json
index 9287100..621167c 100644
--- a/package.json
+++ b/package.json
@@ -74,7 +74,8 @@
     "prettier": "^2.2.1",
     "semantic-release": "^17.3.9",
     "ts-node": "^9.1.1",
-    "vega-lite": "^4.17.0"
+    "vega-lite": "^4.17.0",
+    "vega": "^5.19.1"
   },
   "scripts": {
     "prepublishOnly": "yarn build",
diff --git a/test/vega-lite.test.ts b/test/vega-lite.test.ts
index 91a79c1..b18d9ec 100644
--- a/test/vega-lite.test.ts
+++ b/test/vega-lite.test.ts
@@ -17,10 +17,8 @@ describe("vega-lite", () => {
     it("schema", () => {
         const type = `TopLevelSpec`;
         const config: Config = {
-            path: `node_modules/vega-lite/src/index.ts`,
+            path: "node_modules/vega-lite/src/index.ts",
             type,
-            jsDoc: `none`,
-            extraTags: [],
             encodeRefs: false,
             skipTypeCheck: true,
         };
```

You also need to run this script, though: https://github.com/vega/vega-lite/blob/master/scripts/rename-schema.sh.

**Comment by sramam** [comment]
I see. 
Will it not suffice to cache a snapshot of the generated schema and compare it against that? 
Running a bash script cross-platform from within node seems like more long term pain than necessary. 
However, the dependabot updates to `vega-lite` dev dependency might fail and require a manual intervention. 

Either way, that is step 2. 
This is still stuck at step 1.  The schema generation fails at [this point](https://github.com/vega/ts-json-schema-generator/pull/672/checks?check_run_id=1943868900#step:7:544). I can't tell for sure, but suspect this has something to do with a dev dependency. I was hoping to get some help with that.

**Comment by domoritz** [comment]
My patch adds Vega, which fixes the built for me. 

Yes, we could just copy out the version before running yarn schema in the Vega-Lite directory. In fact, it might be easier to do that in bash and treat it as a separate test (not in jest). What do you think?

Btw, thank you for this pull request. I didn't realize how useful it would be to have this test case.

**Comment by sramam** [comment]
> My patch adds Vega, which fixes the built for me.

I added `vega` and it does seem to fix the issue with generation. Thank you. 

> Yes, we could just copy out the version before running yarn schema in the Vega-Lite directory. In fact, it might be easier to do that in bash and treat it as a separate test (not in jest). What do you think?

I prefer the jest test - since it has two advantages - 
1. Tracks coverage without additional steps - which was my primary motivation here.
2. I run on windows and where possible prefer not to have to deal with bash scripts.

I have made a few changes to the test to avoid the bash script.
1. Generated `build/vega-lite-schema.json` without the post-processing of `rename-schema.sh` (using the `ts-json-schema-generator@0.86.1`) to `test\vega-lite\schema.json` which will serve as the cache. 
2. The `vega-lite` test now also create `test\vega-lite\generated.json`. This might be a temporary measure until this issue is closed.

Basically, this will allow comparing the input the rename-schema.sh - between an official release and the current branch. 
The two are close, but not identical. Any chance I can ask you to take a look? 

> Btw, thank you for this pull request. I didn't realize how useful it would be to have this test case.

I was trying to make my life easier - but you are most welcome. 

Thank you for all the work in maintaining this library. The MappedType is just a beast of a thing to have gotten working to the level you have - makes me feel like a monkey at a keyboard.

**Comment by domoritz** [comment]
> I prefer the jest test - since it has two advantages -

Sounds good to me. 

> Any chance I can ask you to take a look?

Sure. Send me a diff.

**Comment by sramam** [comment]
> Sure. Send me a diff.

SInce PR updates keep only the latest versions, I put these in two files in the PR:

0.86.1: https://github.com/vega/ts-json-schema-generator/blob/5bacc6f843cf34d6f7974eafaa33aa861c22654f/test/vega-lite/schema.json
#672 : https://github.com/vega/ts-json-schema-generator/blob/5bacc6f843cf34d6f7974eafaa33aa861c22654f/test/vega-lite/generated.json

**Comment by domoritz** [comment]
See https://github.com/vega/ts-json-schema-generator/pull/678
